modules/gem5/_posts/2000-02-01-assignment0.md

```diff
@@ -0,0 +1,222 @@
+
+Similar to a real system, a system in gem5 is not complete with only a `board`.
+However, the `board` is the platform through which all the other components communicate with each other.
+In order to build a complete system we need 3 other components: a `processor`, a `cache hierarchy`, and a `memory`.
+This is not exactly the same way you would build a real computer system.
+For real computers, `cache hierarchy` comes pre-packaged with the `processor` and you need other components such as a **power supply** and a **storage drive** to complete a real system. 
+However, there are many research focuses on cache hierarchy design that has motivated this separation of `processor` and `cache hierarchy`.
+
+In addition, you need to specify the clock frequency for the `processor` and `cache hierarchy` to the `board`.
+
+### `gem5.components.processors`
+
+Ready made `processor` objects in the standard library represent the processing cores in a real CPU.
+In this assignment, we are going to use `HW0TimingSimpleCPU`.
+You can find its source code in this repo in `components/processors.py`.
+This processor is based on `SimpleProcessor` from the standard libary.
+Whenever instantiated, it will create a `SimpleProcessor` with **one** `TimingSimpleCPU` core with `RISC-V` instruction set architecture (ISA).
+You can find the source code to `SimpleProcessor` in `gem5/src/python/gem5/components/processors/simple_processor.py`.
+In addition, it is highly recommended that you learn more about gem5's different CPU models in the link below.
+
+- [SimpleCPU](https://www.gem5.org/documentation/general_docs/cpu_models/SimpleCPU)
+- [MinorCPU](https://www.gem5.org/documentation/general_docs/cpu_models/minor_cpu)
+- [O3CPU](https://www.gem5.org/documentation/general_docs/cpu_models/O3CPU)
+
+### `gem5.components.cachehierarchies`
+
+The `cache hierarchy` objects in standard libary are designed to implement **cache coherency protocol** for multi-core processors and model the **interconnect** between multiple cores and multiple memory channels.
+In this assignment we are going to use `HW0MESITwoLevelCache`.
+You can find its source code in `components/cache_hierarchies.py`.
+This cache hierarchy is based on `MESITwoLevelCacheHierarchy` from the standard library.
+You can find the source code to `MESITwoLevelCacheHierarchy` in `gem5/src/python/gem5/components/cachehierarchies/ruby/mesi_two_level_cache_hierarchy.py`.
+Whenever instantiated, a `HW0MESITwoLevelCache` creates a **two level** cache hierarchy with the **MESI** protocol for **cache coherency**.
+It has a **64KiB 8-way set associative L1 instruction cache**, **64KiB 8-way set associative L1 data cache**, **256KiB 4-way set associative L2 unified cache with 16 banks**.
+
+### `gem5.components.memory`
+
+The `memory` objects in standard library are designed to implement various types of single and multi-channel DRAM based memories.
+In this assignment we are going to use `HW0DDR3_1600_8x8`.
```

Comment:
Can you rename this simply `HW0DDR3`?

Reply:
Since there are a lot of places I have used HW0DDR3_1600_8x8 I would prefer to keep it this way. I am worried it might create inconsistencies on how it is referred to.